### PR TITLE
Improve bem-css

### PR DESCRIPTION
**Pull request by tadatuta**

### CODE REVIEW

File: bem-css/index.html [hunk 1]
```diff
@@ -4,18 +4,18 @@
     <title>Holy Grail Markup</title>
     <link rel="stylesheet" href="style.css">
 </head>
-<body>
-<div class="wrapper">
+<body class="page">
+<div class="page__inner">
```
Comment (tadatuta) [inline bem-css/index.html:8]: See https://ru.bem.info/forum/656/ for motivation (in Russian)

File: bem-css/index.html [hunk 2]
```diff
@@ -4,18 +4,18 @@
...
             We specially use 'title' class here instead of 'logo'
-            in order to set a example of problems with the same classnames in different parts of site
+            in order to set an example of problems with the same classnames in different parts of site
             The same block called 'title' also located in div.content.
         -->
         <h1 class="header__title title">CompanyName</h1>
         <form class="header__search search">
             <label class="header__label label">Type to search:
-                <input class="label__field" type="text">
+                <input class="input" type="text">
```
Comment (tadatuta) [inline bem-css/index.html]: semantically input is not just a part of a label

File: bem-css/style.css [hunk 3]
```diff
@@ -76,7 +76,7 @@ body {
   color: #FFF;
 }
 
-.label__field {
+.input {
   display: block;
```
Comment (ihorzenich) [inline bem-css/style.css]: Agree with wrong-named input, regular simple block 'label' shouldn't have any elements at all. 
But regular input shouldn't have so specific rules too!
We need to refactor this naming somehow, maybe `search__field` will be good idea, what do you think?

Comment (ihorzenich) [reply bem-css/style.css]: Great thanks for this PR, @tadatuta!
Can you just fix one small issue with regular `input` styles (see below) and I will merge!
Also, if we already started working on layout 'holder' fixes, maybe you look at BEM-tree for `nav` block and propose how we can separate layout column and navigation?
Thanks!

**Review by ihorzenich — CHANGES_REQUESTED** [review]

File: bem-css/style.css (hunk 3 above)
Comment (tadatuta) [reply bem-css/style.css]: Fixed.

Added proposal for layout separation in second commit: https://github.com/AleshaOleg/holy-grail-markup/pull/17/commits/6fbc90d9a073b9f3f6f6f02fc3f5f58deac251ce

**Review by ihorzenich — APPROVED** [review]

**Comment by ihorzenich** [comment]
Great thanks for this PR, @tadatuta!
Some post-description - what was changed:

- We refactored high-level layout block-naming: add classname `page` to body (block) and change `wrapper` name to some `page__inner` (his element). This helps us keep our BEM-tree stronger and helping avoid possible problems with naming for blocks we can add directly to BODY in future. Before this change we didn't know how to name they, but now we can just simple write something like `page__something` and keep consistency.
- Fix with wrong-named input, cause regular simple block 'label' shouldn't have any elements at all.
- Separate layout column and navigation block (cause `nav` can be placed anyway, not only in left sidebar) 
- Fix BEM-naming for headings - created block heading and realized h2,h3 as it's *modificators*!

**Pull request merged**